### PR TITLE
add util.file_size

### PR DESCRIPTION
**Pull request by dndrks**
### new: util.file_size

adds a `util` function which returns a file's size in megabytes:

```lua
> util.file_size(_path.code.."awake/awake.lua")
0.013152
> util.file_size(_path.audio.."tape/0000.wav")
71.710508
```
- if provided filepath isn't found:
```lua
> util.file_size(_path.audio.."tape/blahblah.wav")
no file found at /home/we/dust/audio/tape/blahblah.wav
<ok>
```
- if no path is provided:
```lua
> util.file_size()
util.file_size requires a path
<ok>
```

### CODE REVIEW

File: lua/lib/util.lua [hunk 1]
```diff
@@ -46,6 +46,26 @@ util.file_exists = function(name)
...
+  if path ~= nil then
+    local f = io.open(path,"r")
+    if f~=nil then
+      local c = f:seek()      -- get current position
+      local s = f:seek("end") -- get file size
+      f:seek("set", c)        -- restore position
+      io.close(f)
+      return s/1000000
+    else
+      print("no file found at "..path)
```
Comment (artfwo) [inline lua/lib/util.lua:62]:
```suggestion
      error("no file found at "..path)
```

File: lua/lib/util.lua [hunk 2]
```diff
@@ -46,6 +46,26 @@ util.file_exists = function(name)
...
+    end
+  else
+    print("util.file_size requires a path")
```
Comment (artfwo) [inline lua/lib/util.lua:65]:
```suggestion
    error("util.file_size requires a path")
```

File: lua/lib/util.lua [hunk 3]
```diff
@@ -46,6 +46,26 @@ util.file_exists = function(name)
...
+--- query file size.
+-- @tparam string name filepath
+-- @treturn number filesize in mb
+util.file_size = function(path)
+  if path ~= nil then
+    local f = io.open(path,"r")
+    if f~=nil then
+      local c = f:seek()      -- get current position
+      local s = f:seek("end") -- get file size
+      f:seek("set", c)        -- restore position
```
Comment (artfwo) [inline lua/lib/util.lua:58]: it's not necessary to seek to the start, if you're closing the file, the first `seek()` call can be also safely omitted.

**Review by artfwo — APPROVED** [review]

File: lua/lib/util.lua [hunk 4]
```diff
@@ -46,6 +46,26 @@ util.file_exists = function(name)
...
+      io.close(f)
+      return s/1000000
```
Comment (artfwo) [inline lua/lib/util.lua:60]: it might make sense to return bytes and let the script author choose the units to use, sometimes MB is 1048576 bytes too :)

**Comment by dndrks** [comment]
@artfwo , thank you for the code review!! folded in your suggestions, updated PR

**Review by artfwo — APPROVED** [review]
good to go!

**Pull request merged**